### PR TITLE
Brave Today: p3a and some small fixes

### PR DESCRIPTION
**Pull request by petemill**
This mainly introduces p3a for Brave Today.

Records the following:

> Did you scroll down on the NTP and see Brave Today?
> No
> Yes
> 
> How many times did you start a new Brave Today session last week?
> None
> 1 session
> 2 to 3 sessions
> 4 to 7 sessions
> 8 to 12 sessions
> 13 to 18 sessions
> 19 to 25 sessions
> More than 25 sessions
> 
> What was the highest number of cards viewed in a Brave Today session?
> None
> 1 card
> 2 to 4 cards
> 5 to 12 cards
> 11 to 20 cards
> 21 to 40 cards
> 41 to 80 cards
> More than 80 cards
> 
> What was the highest number of cards clicked in a Brave Today session?
> None
> 1 card
> 2 to 3 cards
> 4 to 6 cards
> 7 to 10 cards
> 11 to 15 cards
> More than 15 cards

Card counting is achieved by inserting an element at positions 1, 2, and then every 4th +1. Intersection observer then triggers a call to the backend whenever this scrolls in to view. p3a then sends the highest number of scrolled-to cards in a session per week.

Other p3a items are more straightforward, with an addition to `WeeklyStorage` to track a single highest number of the week. Please see commit messages.

Brave Today also now looks at the last 2 weeks of history instead of incorrectly just looking at the default (1 day).

Fixes some minor spacing in the Customize panel with the publisher Toggles and the scrollbar.

Removes `accept-language` header from Brave Today requests. Partially fixes https://github.com/brave/brave-browser/issues/12684 (will need a new Issue created for `user-agent` and `dnt` headers).

Console in debug mode showing scroll triggers for having viewed cards:
image

## Submitter Checklist:

- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by petemill** [comment]
Pushed a test fix (and addition)

**Comment by petemill** [comment]
Adding Issues that this resolves...

**Comment by petemill** [comment]
SonarCloud failure is due to duplication in JSON which cannot be avoided and should be ignored

File: components/brave_extension/extension/brave_extension/background/today/privateCDN.ts [hunk 1]
```diff
@@ -8,6 +8,19 @@ export const URLS = {
...
 }
 
+export async function fetchResource (url: string) {
+  const response = await fetch(url, {
+    // TODO(petemill): strip user-agent when this is possible
+    // https://crbug.com/571722
+    // TODO(petemill): Move this to c++ or a custom c++ privateCDNFetch
+    // API so that we can strip 'dnt' header too.
+    headers: new Headers({
+      'Accept-Language': '*'
```
Comment (fmarier) [inline components/brave_extension/extension/brave_extension/background/today/privateCDN.ts:18]: :+1:

File: components/p3a/brave_p3a_service.cc [hunk 2]
```diff
@@ -73,6 +73,10 @@ constexpr const char* kCollectedHistograms[] = {
     "Brave.NTP.NewTabsCreated",
     "Brave.NTP.SponsoredImagesEnabled",
     "Brave.NTP.SponsoredNewTabsCreated",
+    "Brave.Today.HasEverInteracted",
```
Comment (iefremov) [inline components/p3a/brave_p3a_service.cc:76]: let's keep the list sorted

**Review by iefremov — APPROVED** [review]
C++ part looks good 👍

**Comment by iefremov** [comment]
btw if there are any existing Today/NTP tests, it should be easy do add P3A checks with `HistogramTester`

**Comment by petemill** [comment]
All CI passed apart from macOS which failed due to some known machine failures. Due to that I think we're good to merge

**Comment by petemill** [comment]
CI passed except for a known test failure which was fixed in https://github.com/brave/brave-core/pull/7287

**Comment by bsclifton** [comment]
@petemill don't forget to update https://github.com/brave/brave-browser/wiki/P3A with the questions 😄

**Comment by kjozwiak** [comment]
Is there an associated issue to this PR?

**Comment by petemill** [comment]
Main issue is here https://github.com/brave/brave-browser/issues/12974

I am yet to make follow up issue for the remaining headers to be removed.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.20.17 Chromium: 87.0.4280.88 (Official Build) nightly (64-bit)
-- | --
Revision | 89e2380a3e36c3464b5dd1302349b1382549290d-refs/branch-heads/4280@{#1761}
OS | Windows 10 OS Version 2009 (Build 19042.630)
```

However, both `Brave.Today.WeeklyMaxCardViewsCount` & `Brave.Today.WeeklyMaxCardVisitsCount` are not working and will be addressed via https://github.com/brave/brave-browser/issues/13096 & https://github.com/brave/brave-browser/issues/13097.

**`Case "Brave.Today.HasEverInteracted"`**

* launched `1.20.17 CR: 87.0.4280.88`
* opened a NTP and scrolled down to view Brave Today
* checked `brave://local-state` and ensured that `Brave.Today.HasEverInteracted` is `1`

```
"Brave.Today.HasEverInteracted": {
   "sent": false,
   "value": "1"
```

**`Case "Brave.Today.WeeklySessionCount"`**

* launched `1.20.17 CR: 87.0.4280.88`

Inititally visitng/scrolling Brave Today:

```
"Brave.Today.WeeklySessionCount": {
   "sent": false,
   "value": "1"
```

2-3 Brave Today visit/scrolls:

```
"Brave.Today.WeeklySessionCount": {
   "sent": false,
   "value": "2"
```

4-7 Brave Today visit/scrolls:

```
"Brave.Today.WeeklySessionCount": {
   "sent": false,
   "value": "3"
```

4-7 Brave Today visit/scroll:

```
"Brave.Today.WeeklySessionCount": {
   "sent": false,
   "value": "4"
```

8-12 Brave Today visit/scroll:

```
"Brave.Today.WeeklySessionCount": {
   "sent": false,
   "value": "5"
```

13-18 Brave Today visit/scroll:

```
"Brave.Today.WeeklySessionCount": {
   "sent": false,
   "value": "5"
```

19-25 Brave Today visit/scroll:

```
"Brave.Today.WeeklySessionCount": {
   "sent": false,
   "value": "6"
```

25+ Brave Today visit/scroll:

```
"Brave.Today.WeeklySessionCount": {
   "sent": false,
   "value": "7"
```

**Pull request merged**